src/binary-encoding.adoc

```diff
@@ -0,0 +1,1600 @@
+   └──────────────────┬────────────────────┘
+                 UTF-8 bytes
+----
+
+==== Example encoding of a symbol with 24 bytes of inline text
+[source]
+----
+┌──── Opcode F9 indicates a variable-length symbol with inline text
+│  ┌─── Length: VarUInt 24
+│  │   v  a  r  i  a  b  l  e     l  e  n  g  t  h     e  n  c  o  d  i  n  g
+F9 31 76 61 72 69 61 62 6c 65 20 6c 65 6e 67 74 68 20 65 6e 63 6f 64 69 6e 67
+      └────────────────────────────────┬────────────────────────────────────┘
+                                  UTF-8 bytes
+----
+
+[[symbols_with_symbol_table_addresses]]
+=== Symbols with text in the symbol table
+
+Symbol values whose text can be found in the local symbol table are encoded using opcodes `0xE1` through `0xE3`:
+
+* `0xE1` represents a symbol whose address in the symbol table (aka its symbol ID) is a 1-byte
+<<fixeduint, `FixedUInt`>> that follows the opcode.
+* `0xE2` represents a symbol whose address in the symbol table is a 2-byte <<fixeduint, `FixedUInt`>> that follows
+the opcode.
+* `0xE3` represents a symbol whose address in the symbol table is a <<varuint,`VarUInt`>> that follows the opcode.
+
+Writers MUST encode a symbol address in the smallest number of bytes possible. For each opcode above, the symbol
+address that is decoded is biased by the number of addresses that can be encoded in fewer bytes.
+
+[cols="^1a,1a,1a"]
+|===
+|Opcode |Symbol address range |Bias
+
+|`0xE1`
+|0 to 255
+|0
+
+|`0xE2`
+|256 to 16,640
+|256
+
+|`0xE3`
+|16,641 to infinity
+|16,641
+|===
+
+[[binary_data]]
+== Binary data
+
+[[blobs]]
+=== Blobs
+
+Opcode `FE` indicates a blob of binary data.
+
+==== Example encoding of a 24-byte blob
+[source]
+----
+┌──── Opcode FE indicates a blob, VarUInt length follows
+│   ┌─── Length: VarUInt 24
+│   │
+FE 31 49 20 61 70 70 6c 61 75 64 20 79 6f 75 72 20 63 75 72 69 6f 73 69 74 79
+      └────────────────────────────────┬────────────────────────────────────┘
+                            24 bytes of binary data
+----
+
+
+[[clobs]]
+=== Clobs
+
+Opcode `FF` indicates a clob--character data of an unspecified encoding.
+
+==== Example encoding of a 24-byte clob
+[source]
+----
+┌──── Opcode FF indicates a clob, VarUInt length follows
+│   ┌─── Length: VarUInt 24
+│   │
+FF 31 49 20 61 70 70 6c 61 75 64 20 79 6f 75 72 20 63 75 72 69 6f 73 69 74 79
+      └────────────────────────────────┬────────────────────────────────────┘
+                            24 bytes of binary data
+----
+
+[[containers]]
+== Containers
+
+Each of the container types (list, s-expression, and struct) has both a length-prefixed encoding and a delimited
+encoding.
+
+The length-prefixed encoding places more burden on the writer, but simplifies reading and enables skipping
+over uninteresting values in the data stream. In contrast, the delimited encoding is simpler and faster for
+writers, but requires the reader to visit each child value in turn to skip over the container.
+
+[[lists]]
+=== Lists
```

Comment:
About here I noticed that nulls weren't always being covered in the section for each type.

Reply:
Yeah, I would, so each section has full coverage of the type.